### PR TITLE
Implement native schema inference for PostgreSQL

### LINKED ISSUE #2462: PostgreSQL Native Schema Inference

### PR DESCRIPTION
**Pull request by phillipleblanc**
## 🗣 Description

Implements native schema inference for PostgreSQL.

Also changes to make the integration tests work on the hosted runner with building the binary on `spiceai-runners` and running the tests on `ubuntu-latest-16-cores`.

## 🔨 Related Issues

datafusion-table-providers PR: https://github.com/datafusion-contrib/datafusion-table-providers/pull/151
Closes #2462
Part of #2147

## 📄 Documentation Requirements

N/A

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -126,8 +126,8 @@ datafusion-common = { git = "https://github.com/spiceai/datafusion.git", rev = "
 datafusion-expr = { git = "https://github.com/spiceai/datafusion.git", rev = "fbedf88bb46b3b9b680977a80f76a5ee8d4a9bc4" }
 datafusion-execution = { git = "https://github.com/spiceai/datafusion.git", rev = "fbedf88bb46b3b9b680977a80f76a5ee8d4a9bc4" }
 
-datafusion-federation = { git = "https://github.com/spiceai/datafusion-federation.git", rev = "ce1ce6158ce03f88c67cde25fca00f436c3e1690" }
-datafusion-table-providers = { git = "https://github.com/datafusion-contrib/datafusion-table-providers.git", rev = "2bcf481b4abe9d0bd6bb2479ce49020df66ff97f" }
+datafusion-federation = { git = "https://github.com/spiceai/datafusion-federation.git", rev = "0aabd423ef8aea0ae05479331d1486e9ef8f8061" }
```
Comment (sgrebnov) [inline Cargo.toml:129]: Note: this enables recently added filters pushdown support optimization. I'll ensure latest unparser improvements to correctly support this work are merged today so everything works as expected

File: crates/runtime/tests/duckdb/mod.rs [hunk 2]
```diff
@@ -37,7 +37,18 @@ fn make_test_query(table_name: &str) -> String {
...
+    let sample_csv_contents = include_str!("../test_data/taxi_sample.csv");
+    let sample_json_contents = include_str!("../test_data/taxi_sample.json");
+    // Write the sample file to a temporary directory
+    let temp_dir = std::env::temp_dir().join("spiced_test_data");
+    std::fs::create_dir_all(&temp_dir).expect("failed to create temp dir");
+    let sample_csv_path = temp_dir.join("taxi_sample.csv");
+    std::fs::write(&sample_csv_path, sample_csv_contents).expect("failed to write sample file");
+    let sample_json_path = temp_dir.join("taxi_sample.json");
+    std::fs::write(&sample_json_path, sample_json_contents).expect("failed to write sample file");
+    defer! {
```
Comment (eadgbear) [inline crates/runtime/tests/duckdb/mod.rs:49]: No feedback, this is just a cool crate

File: Cargo.toml (hunk 1 above)
Comment (eadgbear) [reply Cargo.toml:129]: @sgrebnov Out of curiosity are the two connected? Pushdown and native schema inference?

**Comment by Sevenannn** [comment]
Changes look good to me. I triggered a benchmark test on this branch for verifying changes https://github.com/spiceai/spiceai/actions/runs/11505291656.

File: Cargo.toml (hunk 1 above)
Comment (sgrebnov) [reply Cargo.toml:129]: @slyons - `table-providers` crate where new schema inference added depends on `datafusion-federation` where we've added Pushdown optimization and also already included to `table-providers` so any upcoming change in `table-providers` will automatically bring updated version of `datafusion-federation`

**Review by sgrebnov — APPROVED** [review]

**Review by peasee — APPROVED** [review]

**Pull request merged**